### PR TITLE
dataset across namespace with sidecar

### LINKED ISSUE #2061: [FEATURES]Access dataset from pods in different namespace
**What feature you'd like to add:**

Access dataset from pods in different namespace

**Why is this feature needed:**

Fluid uses Kubernetes' native Namespace as a division to manage Datasets on the cloud. Fluid takes a user-defined dataset as the smallest management unit and is responsible for maintaining the data cache life cycle of the dataset. Users can mount PVCs in the container for data access, and obtain higher data access efficiency after data cache acceleration.

In some real scenarios, users will create data-intensive jobs in multiple different namespaces, and these jobs will access the same dataset. For example, multiple data scientists share the same dataset, and each data scientist has its own independent namespace to submit jobs. If the data cache is redeployed for each namespace, it will cause data redundancy and job startup delays.

### PR DESCRIPTION
**Pull request by xliuqq**
### Ⅰ. Describe what this PR does
Support Access dataset from pods in different namespace（Sidecar）

### Ⅱ. Does this pull request fix one issue?

fixes #2061 

### Ⅳ. Describe how to verify it
manual test.

### Ⅴ. Special notes for reviews
currently only verify the AlluxioRuntime.

### CODE REVIEW

File: pkg/ddc/base/runtime_helper.go [hunk 1]
```diff
@@ -53,27 +53,28 @@ func init() {
...
 	}
 	// TODO: create fuse container
 	ds, err := info.getFuseDaemonset()
 	if err != nil {
 		return template, err
 	}
 
-	dataset, err := utils.GetDataset(info.client, info.name, info.namespace)
+	// Note: get the pvc corresponding dataset
+	pvcDataset, err := utils.GetDataset(info.client, pvcName, pvcNamespace)
```
Comment (cheyang) [inline pkg/ddc/base/runtime_helper.go:68]: I suggest renaming `pvcDataset` to `Dataset`?

File: pkg/ddc/thin/referencedataset/cm.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		}
+	default:
+		err := fmt.Errorf("fail to get configmap for runtime type: %s", mountedRuntimeType)
+		return err
+	}
+
+	return nil
+}
+
+func copyConfigMap(client client.Client, srcName string, srcNameSpace string, dstName string, dstNameSpace string, reference metav1.OwnerReference) error {
```
Comment (TrafalgarZZZ) [inline pkg/ddc/thin/referencedataset/cm.go:99]: You can use `types.NamespacedName` instead of `name` and `namespace` for short

Comment (TrafalgarZZZ) [inline pkg/ddc/thin/referencedataset/cm.go:99]: You can use `types.NamespacedName` instead of `name` and `namespace` for short

Comment (TrafalgarZZZ) [inline pkg/ddc/thin/referencedataset/cm.go:99]: Also, I suggest to move the func `copyConfigMap` to a utils package like [`kubeclient.CopySecretToNamespace`](https://github.com/fluid-cloudnative/fluid/blob/master/pkg/utils/kubeclient/secret.go#L58), its's possible we'll reuse this function.

File: pkg/ddc/thin/referencedataset/cm.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	// TODO:  currently the dst configmap name is the same as src configmap name to avoid modify the fuse init container filed,
+	//       but duplicated name error can occurs if the dst namespace has same named runtime.
+	case common.AlluxioRuntime:
+		configMapName := mountedRuntimeName + "-config"
+		err := copyConfigMap(client, configMapName, mountedRuntimeNamespace, configMapName, refNameSpace, ownerReference)
+		if err != nil {
+			return err
+		}
+	case common.JuiceFSRuntime:
+		configMapName := mountedRuntimeName + "-config"
```
Comment (TrafalgarZZZ) [inline pkg/ddc/thin/referencedataset/cm.go:63]: For JuiceFSRuntime, fuse sidecar mutation may need [`mountedRuntimeName.+ "-fuse-script"`](https://github.com/fluid-cloudnative/fluid/blob/master/charts/juicefs/templates/fuse/daemonset.yaml#L178).

File: pkg/ddc/base/runtime_helper.go (hunk 1 above)
Comment (xliuqq) [reply pkg/ddc/base/runtime_helper.go]: fixed

File: pkg/ddc/thin/referencedataset/cm.go (hunk 2 above)
Comment (xliuqq) [reply pkg/ddc/thin/referencedataset/cm.go]: fixed.

Comment (xliuqq) [reply pkg/ddc/thin/referencedataset/cm.go]: fixed.

Comment (xliuqq) [reply pkg/ddc/thin/referencedataset/cm.go]: fixed, move to `kubeclient` package.

File: pkg/ddc/thin/referencedataset/cm.go (hunk 3 above)
Comment (xliuqq) [reply pkg/ddc/thin/referencedataset/cm.go:62]: added, thanks.

**Review by TrafalgarZZZ — APPROVED** [review]

**Review by cheyang — APPROVED** [review]

**Pull request merged**